app/stats/serializers.py

```diff
@@ -121,7 +121,9 @@ def get_result(self, obj):
         if not is_valid:
             return PortInstallation.objects.none()
         today_day = datetime.datetime.now().day
-        last_12_months = datetime.datetime.now(tz=datetime.timezone.utc) - datetime.timedelta(days=int(today_day) + 365)
+        last_12_months = datetime.datetime.now(tz=datetime.timezone.utc) - datetime.timedelta(365)
+        last_12_months.replace(day=1)
```

Comment:
But what about `hour=0, minute=0, second=0, microseconds=0`?
Just thinking loud about alternatives ... For example:
```python
today = datetime.datetime.utcnow().date()
first_day_of_the_month_one_year_ago = datetime.datetime(today.year - 1, today.month, 1, tzinfo=datetime.timezone.utc)
```
That should take care of both leap years and UTC time zone (at least if I understand the first one correctly).

The code to retrieve this weird timestamp is needed twice and sounds complex enough that we could in principle write a simple python function in one of the "common utilities" places and call that function here rather than duplicating the code.

Also, "last_12_months" implies an interval. We need a variable name that implies a fixed timestamp. Anything shorter and clearer than my suggestion would probably be welcome (`one_year_ago` is not accurate, but it might do the job).

(Note: `today_day` is no longer needed.)

Reply:
I still feel the first approach to be more flexible, and more useful if we want to make a utility function. Mostly because we can supply the number of months we want to go back, and then get the first day of that month.

Thank you for pointing out misleading variable names and `hour=0, minute=0, second=0, milliseconds=0`?

Reply:
How many days back will you go in utility function that says "go back for 1 month"?

(Some suggest using `dateutil` for this kind of date manipulations.)

Reply:
Oh, sorry I missed to add the use of `monthDelta` https://pythonhosted.org/MonthDelta/ .

Once this we go back certain months ago in time, and then replace `day=1`.

Reply:
Using `monthDelta` correctly definitely sounds better than manually subtracting 365 days.